felix/bpf/conntrack/cleanup.go

```diff
@@ -288,8 +291,16 @@ func (sns *StaleNATScanner) Check(k KeyInterface, v ValueInterface, _ EntryGet)
 				svcIP = kA
 				svcPort = kAport
 			} else {
-				log.WithFields(log.Fields{"key": k, "value": v}).Error("Mismatch between key and rev key")
-				return ScanVerdictOK // don't touch, will get deleted when expired
+				rv, err := get(revKey)
+				if err != nil {
+					// There is no match for the reverse key, delete it, its useless - we
+					// can get here due to host networked program accessing service
+					// without ctlb when the backed is accessible via tunnel.
+					return ScanVerdictDelete
+				}
+				k = revKey
+				v = rv
+				goto reverse // handle it based on the reverse entry
```

Comment:
Think I'd prefer recursion to a goto (at least that would die with a stack overflow if there was an infinite loop).

Reply:
`reverse` checks what the type is to avoid infinite looping. Will never jump back to this.